extensions/opc/src/opc.cpp

```diff
@@ -465,71 +467,82 @@ std::string nodeValue2String(const NodeData& nd) {
       ret_val = std::string(reinterpret_cast<const char *>(value.data), value.length);
       break;
     }
-    case UA_DATATYPEKIND_BOOLEAN:
-      bool b;
+    case UA_DATATYPEKIND_BOOLEAN: {
+      bool b = false;
       memcpy(&b, nd.data.data(), sizeof(bool));
       ret_val = b ? "True" : "False";
       break;
-    case UA_DATATYPEKIND_SBYTE:
-      int8_t i8t;
+    }
+    case UA_DATATYPEKIND_SBYTE: {
+      int8_t i8t = 0;
       memcpy(&i8t, nd.data.data(), sizeof(i8t));
       ret_val = std::to_string(i8t);
       break;
-    case UA_DATATYPEKIND_BYTE:
-      uint8_t ui8t;
+    }
+    case UA_DATATYPEKIND_BYTE: {
+      uint8_t ui8t = 0;
       memcpy(&ui8t, nd.data.data(), sizeof(ui8t));
       ret_val = std::to_string(ui8t);
       break;
-    case UA_DATATYPEKIND_INT16:
-      int16_t i16t;
+    }
+    case UA_DATATYPEKIND_INT16: {
+      int16_t i16t = 0;
       memcpy(&i16t, nd.data.data(), sizeof(i16t));
       ret_val = std::to_string(i16t);
       break;
-    case UA_DATATYPEKIND_UINT16:
-      uint16_t ui16t;
+    }
+    case UA_DATATYPEKIND_UINT16: {
+      uint16_t ui16t = 0;
       memcpy(&ui16t, nd.data.data(), sizeof(ui16t));
       ret_val = std::to_string(ui16t);
       break;
-    case UA_DATATYPEKIND_INT32:
-      int32_t i32t;
+    }
+    case UA_DATATYPEKIND_INT32: {
+      int32_t i32t = 0;
       memcpy(&i32t, nd.data.data(), sizeof(i32t));
       ret_val = std::to_string(i32t);
       break;
-    case UA_DATATYPEKIND_UINT32:
-      uint32_t ui32t;
+    }
+    case UA_DATATYPEKIND_UINT32: {
+      uint32_t ui32t = 0;
       memcpy(&ui32t, nd.data.data(), sizeof(ui32t));
       ret_val = std::to_string(ui32t);
       break;
-    case UA_DATATYPEKIND_INT64:
-      int64_t i64t;
+    }
+    case UA_DATATYPEKIND_INT64: {
+      int64_t i64t = 0;
       memcpy(&i64t, nd.data.data(), sizeof(i64t));
       ret_val = std::to_string(i64t);
       break;
-    case UA_DATATYPEKIND_UINT64:
-      uint64_t ui64t;
+    }
+    case UA_DATATYPEKIND_UINT64: {
+      uint64_t ui64t = 0;
       memcpy(&ui64t, nd.data.data(), sizeof(ui64t));
       ret_val = std::to_string(ui64t);
       break;
-    case UA_DATATYPEKIND_FLOAT:
+    }
+    case UA_DATATYPEKIND_FLOAT: {
       if (sizeof(float) == 4 && std::numeric_limits<float>::is_iec559) {
-        float f;
+        float f = NAN;
```

Comment:
I'd prefer to not bring NaNs to the mix, and just zero-initialize floating point types, like everything else.

Reply:
I think you are right, `NAN` was suggested by clang-tidy but using `0` is more consistent, updated in 760f6ab8125f79b5454684d6519442d725690a5c